packages/cardpay-sdk/providers/json-rpc-provider.ts

```diff
@@ -0,0 +1,170 @@
+import { BigNumber, logger, providers } from 'ethers';
```

Comment:
Can we import this file from ethers somehow? I suppose it's copied from https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/json-rpc-provider.ts

Reply:
I updated the way this provider handles error messages because we use error messages to calculate safe addresses and estimate execution gas.

Reply:
In that case I think it would be good to add a comment with that the file was copied and modified. Or even better, commit the copied file and then commit the change so it's clearer what we copied and what we changed

Reply:
Added comments and commits provider modifications